### DISCUSSION TITLE
Column header custom render

### DISCUSSION THREAD

**Discussion by GuiPretto** [icflorescu/mantine-datatable#126] [Ideas]
Hi! I'm currently using this lib and a feature I really miss is the capability of rendering a custom header for the columns.

So far we can render the column body cells, and adding the column header cell to this group would make this lib even better.

I can do the code, since it's a pretty simple thing. And talking about that, I was thinking of naming this prop as "renderHeader",since "render" already exists for rows cells.

**Comment by icflorescu** [comment]
Hi!

I believe you can actually render a custom header, since the column's title accepts a `ReactNode`.
True, it **doesn't accept a function** returning a `ReactNode`, but I can't think of a use-case for it. What would its parameters be?

If you can think of a possible usage scenario where you'd need a dynamic title, I guess we could simply refactor the `title` to accept a `ReactNode` **or a function returning `ReactNode`** instead of adding a new method.

**Reply by GuiPretto** [reply]
Hey, sorry for not seeing it correctly, since there was no example and the prop was named only "title" I completely forgot to see its type (even though it is right above the render one lol).

I'm gonna make a PR changing it to accept a function as well.

**Reply by GuiPretto** [reply]
Actually, thinking more about it, I don't really see the need to accept a function, only accepting a ReactNode seems enough.
